torch/_refs/__init__.py

```diff
@@ -4619,20 +4619,29 @@ def logspace(
         if isinstance(end, FloatLike):
             end = sym_int(end)
 
+    if py_any(isinstance(arg, complex) for arg in (start, end, steps)):
+        default_complex_dtype = utils.corresponding_complex_dtype(
+            torch.get_default_dtype()
+        )
+        dtype = default_complex_dtype
+        _dtype = None  # torch.linspace will update the correct dtype
```

Comment:
Doesn't linspace also make it the `default_complex_dtype`? Is there any difference between passing in None and passing in dtype?

Reply:
They are equivalent, we can do either way.